### PR TITLE
setup preview / testnet deployments

### PR DESCRIPTION
**Pull request by sebastianscatularo**
- move build to github pages
- enable two deployment flavors of the same source code
  - preview (mainnet)
  - testnet  (testnet)

### CODE REVIEW

**Review by raop155 — APPROVED** [review]
LGTM 🚀

File: .github/workflows/preview.yml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        run: |
+          sleep 30s
+          mkdir -p preview/${{ env._BRANCH_NAME_SHA }}
+          pushd preview/${{ env._BRANCH_NAME_SHA }}
+          [[ ! -f .latest ]] && touch .latest
+          [[ ! -f .artifacts ]] && touch .artifacts
+          cat .latest >> .artifacts
+          echo ${{ env._PKG_VERSION }} > .latest
+          git add .artifacts .latest
+          git commit -m"Update preview artifacts ${{ env._PKG_VERSION }}"
```
Comment (valentinoConti) [inline .github/workflows/preview.yml:127]: space missing here too

Comment (raop155) [reply .github/workflows/preview.yml:127]: Good eyes 👀 , but it does not affect tho :)

File: .github/workflows/cleanup.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+          git config --global user.email "devops@xlabs-playground.xyz"
+      - name: Set Version
+        run: echo "_BRANCH_NAME_SHA=$(echo ${{ github.event.pull_request.head.ref }} | sha256sum | cut -c -10)" >> "${GITHUB_ENV}"
+      - name: Unpublish and remove preview artifacts
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          npm run unpublish ${_BRANCH_NAME_SHA}
+          git add ${_BRANCH_NAME_SHA}
+          git commit -m"Remove preview artifacts"
```
[the remark this answers is not in the document]
Comment (sebastianscatularo) [reply .github/workflows/cleanup.yml:31]: does not affect

**Pull request merged**